src/components/forms/CollisionTileLabelsPicker.tsx

```diff
@@ -0,0 +1,280 @@
+        icon = <BrushToolbarTileSlope22LeftTopIcon $color={value.color} />;
+        break;
+      case "slope_22_left_bot":
+        name = l10n("FIELD_COLLISION_SLOPE_22_LEFT_BOT");
+        icon = <BrushToolbarTileSlope22LeftBottomIcon $color={value.color} />;
+        break;
+      case "spare_08":
+        name = l10n("FIELD_COLLISION_SPARE", { tile: 8 });
+        icon = (
+          <BrushToolbarExtraTileIcon
+            $value={COLLISIONS_EXTRA_SYMBOLS[0]}
+            $color={value.color}
+          />
+        );
+        break;
+      case "spare_09":
+        name = l10n("FIELD_COLLISION_SPARE", { tile: 9 });
+        icon = (
+          <BrushToolbarExtraTileIcon
+            $value={COLLISIONS_EXTRA_SYMBOLS[1]}
+            $color={value.color}
+          />
+        );
+        break;
+      case "spare_10":
+        name = l10n("FIELD_COLLISION_SPARE", { tile: 10 });
+        icon = (
+          <BrushToolbarExtraTileIcon
+            $value={COLLISIONS_EXTRA_SYMBOLS[2]}
+            $color={value.color}
+          />
+        );
+        break;
+      case "spare_11":
+        name = l10n("FIELD_COLLISION_SPARE", { tile: 11 });
+        icon = (
+          <BrushToolbarExtraTileIcon
+            $value={COLLISIONS_EXTRA_SYMBOLS[3]}
+            $color={value.color}
+          />
+        );
+        break;
+      case "spare_12":
+        name = l10n("FIELD_COLLISION_SPARE", { tile: 12 });
+        icon = (
+          <BrushToolbarExtraTileIcon
+            $value={COLLISIONS_EXTRA_SYMBOLS[4]}
+            $color={value.color}
+          />
+        );
+        break;
+      case "spare_13":
+        name = l10n("FIELD_COLLISION_SPARE", { tile: 13 });
+        icon = (
+          <BrushToolbarExtraTileIcon
+            $value={COLLISIONS_EXTRA_SYMBOLS[5]}
+            $color={value.color}
+          />
+        );
+        break;
+      case "spare_14":
+        name = l10n("FIELD_COLLISION_SPARE", { tile: 14 });
+        icon = (
+          <BrushToolbarExtraTileIcon
+            $value={COLLISIONS_EXTRA_SYMBOLS[6]}
+            $color={value.color}
+          />
+        );
+        break;
+      case "spare_15":
+        name = l10n("FIELD_COLLISION_SPARE", { tile: 15 });
+        icon = (
+          <BrushToolbarExtraTileIcon
+            $value={COLLISIONS_EXTRA_SYMBOLS[7]}
+            $color={value.color}
+          />
+        );
+        break;
+      default:
+        icon = <div />;
+    }
+  }
+
+  return (
+    <StyledCollisionTileLabelPicker>
+      <StyledCollisionTileIcon>{icon}</StyledCollisionTileIcon>
+      <Input
+        id={value.key + ".name"}
+        value={value.name ?? ""}
+        placeholder={name}
+        onChange={(e) => {
+          onChange({
+            key: value.key,
+            name: e.currentTarget.value,
+            icon: value.icon,
+            color: value.color,
+          });
+        }}
+      />
+      <StyledColorInput
+        // type="color" //truncates alpha value
```

Comment:
`type="color"` gives you the nice color picker, but it truncates the alpha value. Without transparency, the extra tiles look like they're replacing a solid tile instead of being placed in addition to it. It also means I can't make some tiles really faint when they're less important. Killing this and reverting some of the styling was the quick and easy fix, but perhaps there's a way to style the color picker popup and configure it to allow alpha values? If not, maybe a separate opacity slider.

Reply:
Hmm, yeah I noticed that switching to the default colour picker also removed the alpha channel for each tile. Personally I didn't mind as I found it confusing to have the two alphas for each tile (the per tile and the global one) and wasn't sure why setting to full alpha on the settings page wasn't showing as I expected in the world view (because I didn't notice the global setting) but if that feature is important to you we can keep it.

I think I might adapt this PR to work as I described above instead though:
- Keep the collision layer opacity setting placed in a panel just below the brush toolbar
- Move the data to be defined in the `engine.json` file rather than exposed in the IDE on the settings page
- I'll can allow colours defined in with alpha channels 

But I think this could be an opportunity to allow the collisions tiles to be much more configurable, so in the engine.json
- allow specifying exactly which bits are set
- allow setting which collision tiles are enabled on a per scene type basis, e.g. ladders only make sense when using platform scenes so don't show them in the toolbar for anything else and that value could be reused for something else on other scene types

I need to think on this a bit more before attempting

Reply:
I've extracted the layer opacity portion of this PR and merged (+credited you in changelog) here 0a88cfacd1709b8c64e2e836d0fb62f070138fb8. Will pick this up again next week to have think how best to handle the collision tile replacement portion 👍

Reply:
You could use mask and set bits. COLLISION_BOTTOM, for example, sets the lower nibble to 2 (ie.  (tile & 0xF0) | 0x02). It's mask is 0xF0 and its set bits are 0x02. This could also be stored as a single string, XXXX0010. When you draw your brush toolbar, you group buttons by similar mask bits. Mask and flag bits for the current tile layers looks like this:

```
solid              XXXX1111
top                XXXX0001
bottom             XXXX0010
left               XXXX0100
right              XXXX1000
ladder             0001XXXX
slope_45_right     00101000
slope_45_left      00110100
slope_22_right_bot 01110010
slope_22_right_top 01111010
slope_22_left_top  01110110
slope_22_left_bot  01110010
spare_08           1000XXXX
spare_09           1001XXXX
spare_10           1010XXXX
spare_11           1011XXXX
spare_12           1100XXXX
spare_13           1101XXXX
spare_14           1110XXXX
spare_15           1111XXXX
```

It's clean, groups the default tiles the same way, and allows you to ditch the "extra" collision data for slopes. Might even go so far as to name those mask groups and give each of them an opacity slider, removing the need for per-tile opacity.

Reply:
I went ahead and implemented this. No more customization in settings. Added to engine.json in sceneTypes like this; 
`{
      "key": "METROIDVANIA",
      "label": "GAMETYPE_METROIDVANIA",
      "files": ["include/states/metroidvania.h", "src/states/metroidvania.c"],
      "collisionTileLabels": [
        { "flag": 15, "mask": 15, "color": "#FF0000FF", "icon": "█", "name": "FIELD_SOLID" },
        { "flag": 1, "mask": 1, "color": "#FF0000FF", "icon": "▘", "name": "FIELD_COLLISION_TOP_LEFT" },
        { "flag": 2, "mask": 2, "color": "#FF0000FF", "icon": "▝", "name": "FIELD_COLLISION_TOP_RIGHT" },
        { "flag": 4, "mask": 4, "color": "#FF0000FF", "icon": "▖", "name": "FIELD_COLLISION_BOTTOM_LEFT" },
        { "flag": 8, "mask": 8, "color": "#FF0000FF", "icon": "▗", "name": "FIELD_COLLISION_BOTTOM_RIGHT" },
        { "flag": 16, "mask": 16, "color": "#00A000FF", "icon": "H", "name": "FIELD_LADDER" },
        { "flag": 32, "mask": 32, "color": "#80808080", "icon": "+", "name": "FIELD_PRIORITY" },
        { "flag": 64, "mask": 64, "color": "#2040F040", "icon": " ", "name": "FIELD_PROP_WATER" },
        { "flag": 128, "mask": 128, "color": "#F080F080", "icon": "s", "name": "FIELD_SCRIPT" }
      ]
    },`